include/stdx/tuple.hpp

```diff
@@ -486,6 +486,18 @@ constexpr auto for_each(Op &&op, T &&t) -> Op {
         return op;
     });
 }
+
+template <typename... Ts>
+class one_of : public detail::tuple_impl<std::index_sequence_for<Ts...>,
+                                         detail::index_function_list<>, Ts...> {
+    template <typename T>
+    constexpr friend auto operator==(one_of const &lhs, T const &rhs) -> bool {
+        return lhs.apply(
+            [&](auto &&...args) { return ((args == rhs) || ...); });
+    }
+};
+template <typename... Ts> one_of(Ts...) -> one_of<Ts...>;
```

Comment:
suggestion:  I think the tuple should be constructed from `Ts const &`. It want's to be a view.

Reply:
Ben and I discussed this and for our current use cases we just need built-in value types. Keep this solution for now and consider adding non-movable/non-copyable view in the future.